### ISSUE TITLE
qt wallet

### ISSUE THREAD

**Issue by nightrid3r** [forknote/forknote#1]
can you make a forknote version of the bytecoin/cryptonote gui wallet?

**Comment by pmitchev** [comment]
It is in the plans, but there is not a specific schedule for it. It depends on time/money available and the interest in it.

**Comment by cycleworm** [comment]
any updates in that?

**Comment by pmitchev** [comment]
Currently not in progress. There is 50 000 DSH bounty for it if anybody is interested.

**Comment by cycleworm** [comment]
Hi pmitchev,
thanks for the quick response. 

oh, a pity that it is not in progress. but nice that a bounty is offered.

is there somewhere a detailed Definition of the requirements?

**Comment by pmitchev** [comment]
* Bytecoin Wallet clone (https://github.com/amjuarez/bytecoin-gui)
* Ability to add/edit/delete Forknote configs
* Ability to run multiple Forknote configs simultaneously
* Ability to do everything Bytecoin Wallet is able to do for every Forknote config
* Opensource (that's obvious, because of the Bytecoin wallet license)

That's about it.

**Comment by cycleworm** [comment]
cool. thanks for the info.

**Comment by lanaiscool** [comment]
anypne working on this ?

**Comment by zhmrab** [comment]
I am going to start working on that and will share the progress as i go along :)

**Comment by lanaiscool** [comment]
Nicee im working on a web wallet alrewdy made a web wallet for another coin called gulden still working on the web wallet for forknote coins only problem is that there is nog simple library for it and the rpc daemon only support named parameters so need to write something custom

**Comment by zhmrab** [comment]
Agreed! Do you have a repo for the forknote web wallet you're currently working on?

named parameters are indeed an issue but I'm looking into it :)

**Comment by zhmrab** [comment]
@lanaiscool I'm trying to use the generator to compile binaries that don't need named parameters with the .json version of my .conf Forknote file that is on my servers, and I'm getting this : 

ERROR WRONG AGENT CONNECTED TO THE NETWORK! Closing connection.

The generator operated smoothly but I may be missing something~ If I figure this out then making a GUI is going to be a lot easier, since you dont need to specify config file anymore, 

Can you actually use the generator for make files that will connect to forknote servers?

**Comment by prashantprabhakar** [comment]
@zhmrab Good to hear someone is working on this. Good luck (y)

**Comment by VluCash** [comment]
@lanaiscool  Will you be sharing the code to the web wallet?

@zhmrab  How is the GUI wallet coming along?

**Comment by Remco75** [comment]
@lanaiscool any progress on the web-wallet? Can I help out / is there a repo already? this should be do-able I guess

**Comment by VluCash** [comment]
I’m also interested in a web wallet

**Comment by lanaiscool** [comment]
Well have a problem with sending a transaction my request looks right but i get destination empty as error code when that is fixed it is considered done

**Comment by Remco75** [comment]
ok, if we can help out, plz let me know :-)

**Comment by fuegohd** [comment]
Looking to get this GUI made in qt or python. I'll update here when it's ready and keep checking back to see if someone beats me to it.

**Comment by lanaiscool** [comment]
@siliconlumen do you have that wallet on github ? i can take a look i have made a web wallet for forknote coin which is almost finished i can send coins check balance etc

**Comment by Remco75** [comment]
Very interested in the wallet as well! will help out where I can

**Comment by siliconlumen** [comment]
After a weekend of debugging I am without a clue how to solve the qt wallet problem...
Setup: 1 daemon VPS Ubuntu server, P2P: xx253, RPC: xx254
2 seed nodes VPS Ubuntu server, listening on P2P: xx253
The daemon runs fine, I can open the Simplewallet and send/receive coins on the server and transfer to an external Win PC running the forknoted+simplewallet. I can also mine from this external WinPC.
So that's ok I guess.
Problems:
°The QT wallet tries to connect to the daemon server on RPC: xx254, **Auto Selection**. The wallet core logs an error 'Failed to invoke Command_Handshake' and IP of one seed node and the P2P xx253 port (Outgoing). The wallet hangs on Synchronising.
° When I try to use the daemon IP and RPC xx254 port the wallet is synchronised (Remote daemon manually), shows the correct BC_height and is connected to my 2 peers. I can't click on the Blockchain for more info. The wallet logs shows a 'Failed to query blocks, NodeErrorCatergory5, internal node error'.
![afbeelding](https://user-images.githubusercontent.com/35366148/36076051-8e91ed9c-0f57-11e8-9df4-6bcbf3d21129.png)
For this qt wallet I used a clone from forknote/intensecoin and changed the things I thought were needed. I will attach a daemon log from the moment the wallet is running and the daemon tries to connect to my wan-ip and P2P port xx253.
[daemon_log.txt](https://github.com/forknote/forknote/files/1714638/daemon_log.txt)

I will try to upload the sources on github, probably later today or tomorrow.
I know it's difficult to debug this and maybe it can't work due to incompatibility issues between the forks.

Thanks,

**Comment by siliconlumen** [comment]
Eureka ! Well kind off...
The QT Wallet is connected and I can receive coins :)
Next question: the wallet runs fine on a win pc running VS2015, how can I make a runtime .exe file that runs on any pc ? I have tried to add the qt dll en vs runtime dll files but the wallet doesn't run and gives an exception error...

![afbeelding](https://user-images.githubusercontent.com/35366148/36254830-d69df34a-124c-11e8-8a46-0464e6a0b42a.png)

**Comment by lanaiscool** [comment]
@siliconlumen what files are in the debug folder of your project

**Comment by siliconlumen** [comment]
I have these in my Debug folder, copied them from the msvc2015 folder (NOT 64-bit folder) and the wallet runs but only on this PC. On another pc I get or nothing or this message: **qt platformplugin windows not found in ""**

![afbeelding](https://user-images.githubusercontent.com/35366148/36265382-684dd95a-126f-11e8-9286-bc095c10cb36.png)

**Comment by lanaiscool** [comment]
@siliconlumen so you mentioned you tried importing the dlls that are missing you also tried to build a release directly ?
Maybe this will help https://stackoverflow.com/questions/14932315/how-to-compile-qt-5-under-windows-or-linux-32-or-64-bit-static-or-dynamic-on-v

**Comment by mtl1979** [comment]
@siliconlumen Under the directory containing the executable, you need to have directory "platforms" containing file "qwindows.dll", you can copy it from "qtbase\plugins\platforms" under the directory containing Qt 5.x sources. If you have binary-only version of Qt 5.x, the file should be inside "plugins\platforms" directory under the directory where the binary-only version is installed.

**Comment by siliconlumen** [comment]
I have made some progress with windeployqt from the QT install dir. It generates a bunch of files and folders and the wallet seems to run on most pc's now. Still in Debug mode, I will test this in Release mode the coming days.

**Comment by siliconlumen** [comment]
When I configure the QT wallet as Auto it refuses to sync and gives me this error in the core.log file:
**'Failed to validate block c4714bf617df90d48b686019f45873acab4f24ded55e2d4b58628991d88257f9: Wrong block version'**
If I change the wallet setting to my daemon server IP and RPC port the wallet does 'sync' but my local blocks.bin file doesn't update. Anyone have an idea ?

**Comment by mtl1979** [comment]
Local blocks.bin doesn't update when using remote daemon as the blockchain is stored inside the wallet file.
I would double-check the internal config variables in the Qt wallet and compare them with the ones in daemon server.

**Comment by Bitcoin-N** [comment]
I am in the same point gui generated, it does not synchronize and as soon as I change my coin to genesis_coinbase_tx_hex, the program closes.

Is there gui that supports forknote?

**Comment by DavidSimalDelVal** [comment]
I hope forknote developers improve their own code

**Comment by Bitcoin-N** [comment]
from what I see it is a matter of the core

**Comment by Bitcoin-N** [comment]
after a lot of fighting with the issue.

1.- compile wallet with an optional GENESIS_COINBASE_TX_HEX, generated with --print-genesis-tx
2.- I enter options and change to one of the options, local daemon or remote daemon.
3.- I start to compile the wallet but this time with the original GENESIS_COINBASE_TX_HEX and the good of the currency in question.
4.- Start the wallet and show the current blocke.

the wallet works but does not download the blockchain.

**Comment by Bitcoin-N** [comment]
\cryptonote\src\CryptoNoteCore\BlockchainIndices.cpp
\cryptonote\src\CryptoNoteCore\BlockchainIndices.h
\cryptonote\src\CryptoNoteCore\BlockIndex.cpp
\cryptonote\src\CryptoNoteCore\BlockIndex.h
\cryptonote\src\CryptoNoteCore\CoreConfig.cpp
\cryptonote\src\CryptoNoteCore\CoreConfig.h
\cryptonote\src\CryptoNoteCore\Blockchain.cpp
\cryptonote\src\CryptoNoteCore\Blockchain.h
\cryptonote\src\CryptoNoteCore\IBlock.cpp
\cryptonote\src\CryptoNoteCore\IBlock.h
\cryptonote\src\BlockchainExplorer/BlockchainExplorerDataBuilder.cpp
\cryptonote\src\BlockchainExplorer/BlockchainExplorerDataBuilder.h

**Comment by Bitcoin-N** [comment]
uploaded code as it works

https://github.com/Bitcoin-N/Bitcoinnova-Gui-BETA.git

**Comment by Bitcoin-N** [comment]
https://github.com/Bitcoin-N/Bitcoinnova-Gui-BETA/releases/download/1.0/Bitcoinnova-Gui-BETA-Win64.rar

after a lot of fighting with the issue.

1.- compile wallet with an optional GENESIS_COINBASE_TX_HEX, generated with --print-genesis-tx
2.- I enter options and change to one of the options, local daemon or remote daemon.
3.- I start to compile the wallet but this time with the original GENESIS_COINBASE_TX_HEX and the good of the currency in question.
4.- Start the wallet and show the current blocke.

the wallet works but does not download the blockchain.


\cryptonote\src\CryptoNoteCore\BlockchainIndices.cpp
\cryptonote\src\CryptoNoteCore\BlockchainIndices.h
\cryptonote\src\CryptoNoteCore\BlockIndex.cpp
\cryptonote\src\CryptoNoteCore\BlockIndex.h
\cryptonote\src\CryptoNoteCore\CoreConfig.cpp
\cryptonote\src\CryptoNoteCore\CoreConfig.h
\cryptonote\src\CryptoNoteCore\Blockchain.cpp
\cryptonote\src\CryptoNoteCore\Blockchain.h
\cryptonote\src\CryptoNoteCore\IBlock.cpp
\cryptonote\src\CryptoNoteCore\IBlock.h
\cryptonote\src\BlockchainExplorer/BlockchainExplorerDataBuilder.cpp
\cryptonote\src\BlockchainExplorer/BlockchainExplorerDataBuilder.h



create the file "bitcoinnovawallet.cfg" in the folder C: \Users\YOUR-USER\AppData\Roaming\bitcoinnova or edit archiva

with the following content:


"connectionMode": "remote",
     "currentPool": "pool.bitcoinn.biz:11133",
     "daemonPort": 45223,
     "miningPools": [
         "pool.bitcoinn.biz:11133",
         "bitcoinn.multipool.es:3333",
         "bitcoinn.criptominando.eu:11133",
         "btn.dreampool.info:3760",
         "superblockchain.con-ip.com:11133",
         "btn.gonspool.com:11133"
     ],
     "miningThreads": 10,
     "recentWallets": [
         "C: /Users/piter/AppData/Roaming/bitcoinnova/bitcoinnovawallet.wallet"
     ],
     "remoteNode": "85.214.158.156:45223",
     "remoteNodes": [
         "85.214.158.156:45223",
         "37.59.108.203:45223"
     ],
     "tracking": false
}

**Comment by nightfury626** [comment]
i need help please guys. I built my coin, seed nodes and all. how do i put together a qt wallet for linux, mac and windows users to where they can download it in exe or zip files? I'm mostly concerned about windows since thats what majority of people use. can I build one with linux or do i need a windows computer to build a windows zip or exe wallet. I need a step by step instruction please and thank you.

**Comment by blueavenue** [comment]
Any news about the Wallet?

**Comment by intrinsiccoin** [comment]
You can look at my repo for windows ONLY GUI wallet. The build is not straightforward though, unfortunately. 

Do the following:

1. Fork the [IntrinsicCoin-GUI-Wallet](https://github.com/intrinsiccoin/IntrinsicCoin-GUI-Wallet.git) repository, add the submodule in your clone or just replace the 'IntrinsicCoin' folder with your coin code.
2. BEFORE building the wallet you need to do the following:
     a. Delete the contents of the 'libs' folder within the GUI wallet root directory.
     b. In the IntrinsicCoin-GUI-Wallet/IntrinsicCoin/External folder create a 'Build' folder, execute cmake with DSTATIC option 0. This step is important.
     c. Build in MSVCS 2017/Boost 1.66, then copy the resulting rocksdblib.lib file from the 'Build' folder into the 'libs' folder within the GUI wallet root directory.
     d. Next, Build the 'IntrinsicCoin' using cmake with DSTATIC option 0. Again, this step is important. Note: The build in MSVC 2017 may fail but dont worry, just copy the resulting '.lib' files from your coin's build folder into the 'libs' folder within the GUI wallet root directory.
     c. Next get into the GUI wallet root directory, ie 'IntrinsicCoin-GUI-Wallet' and do cmake with DSTATIC option 1, and build your wallet. Hopefully you will see the 'IntrinsicCoinWallet.exe' file your build folder. 
Now go ahead and try out your wallet.

So in short, this is a crude/tricky way to get your code working on Windows. But credit is due to @Jasin. Thanks to his work, this wallet was possible.

But its way too easy to get your coin a GUI wallet for Ubuntu. Fork the ChavezCoinWallet code, put your coin's code in the 'cryptonode' folder and execute cmake/make. Voila, your wallet is ready. Its pretty straightforward. Hope this helps in your first trial and error.

**Comment by Vordas** [comment]
WARNING: do not waste time on intrinsic wallet - the icons do not show up and it's code is absolutely butchered to a degree beyond saving.

**Comment by Remco75** [comment]
@Vordas , thnx for the warning. But i have a gentle suggestion. Next time you comment, maybe be a bit nicer to the developer less experienced that yourself.... He's trying to help out, so if you have suggestions for improvements in the code, please do give them. Or at least point out WHY the code is beyond saving. That way we all might learn a thing or 2.

**Comment by Vordas** [comment]
@Remco75 My apologies. You are right. If I make some additional progress, I'll post another comment about it.

**Comment by kennetanti** [comment]
Is the 50000 dash bounty still available?

**Comment by dedrisproject** [comment]
I working on final version of gui wallet

**Comment by pmitchev** [comment]
yes, it's still available. 50000 DSH on the democats chain

**Comment by Bitminereu** [comment]
waiting for final version of gui wallet.

**Comment by Emojibit** [comment]
any updates @Kaosevil

**Comment by bigstunta101** [comment]
@intrinsiccoin thanks for the step by step direction but unfortunately, i'm using linux and dont currently have any windows computer. so my main problem was trying to build a windows wallet using linux OS. unless that's not possible

**Comment by kennetanti** [comment]
@bigstunta101 using a linux build of mingw could possibly work. Although it looks like he is using vs compilers so your best bet is probably to use a windows VM...if it builds with gcc for linux then it probably will work with mingw for windows
